spec/models/manageiq/providers/ibm_cloud/power_virtual_servers/storage_manager/cloud_volume_spec.rb

```diff
@@ -0,0 +1,27 @@
+describe ManageIQ::Providers::IbmCloud::PowerVirtualServers::StorageManager::CloudVolume do
+
+  let(:ems_cloud) { FactoryBot.create(:ems_ibm_cloud_power_virtual_servers_cloud) }
+  let(:ems_storage) { FactoryBot.create(:ems_ibm_cloud_power_virtual_servers_storage, :parent_ems_id => ems_cloud.id) }
+  let(:cloud_volume) { FactoryBot.create(:cloud_volume_ibm_cloud_power_virtual_servers, :ext_management_system => ems_storage ) }
+
+
+  context "#validate_delete_volume" do
+    it "status is in-use" do
+      expect(cloud_volume).to receive(:status).and_return("in-use")
```

Comment:
`:status` is a column on the `cloud_volumes` table so instead of mocking this you can set the `:status` attribute as part of your `let(:cloud_volume)` above

You can use contexts to do something like

```
let(:cloud_volume) { FactoryBot.create(:cloud_volume_ibm_cloud_power_virtual_servers, :ext_management_system => ems_storage, :status => status) }
describe "#validate_delete_volume" do
  context "status is in-use" do
    let(:status) { "in-use" }

    it "validates unavailable" do
    end
  end

  context "status is available"
    let(:status) { "available" }

    it "validates available" do
    end
  end
end
```

The reason I don't like to mock stuff like this is it can hide things that might end up failing at runtime but be passing specs.  Theoretically if we renamed the `CloudVolume#status` column to something else this spec would still pass but it would blow up at runtime.